unstructured/partition/docx.py

```diff
@@ -173,20 +187,38 @@ def partition_docx(
         to .docx before partition. We want the original source filename in the metadata.
     metadata_last_modified
         The last modified date for the document.
+    languages
+        User defined value for `metadata.languages` if provided. Otherwise language is detected
+        using naive Bayesian filter via `langdetect`. Multiple languages indicates text could be
+        in either language.
+        Additional Parameters:
+            detect_language_per_element
+                Detect language per element instead of at the document level.
     """
     # -- verify that only one file-specifier argument was provided --
     exactly_one(filename=filename, file=file)
 
-    return list(
-        _DocxPartitioner.iter_document_elements(
-            filename,
-            file,
-            metadata_filename,
-            include_page_breaks,
-            metadata_last_modified,
+    if not isinstance(languages, list):
+        raise TypeError(
+            'The language parameter must be a list of language codes as strings, ex. ["eng"]',
+        )
+
+    elements = list(
+        apply_lang_metadata(
+            elements=_DocxPartitioner.iter_document_elements(
+                filename,
+                file,
+                metadata_filename,
+                include_page_breaks,
+                metadata_last_modified,
+            ),
+            languages=languages,
+            detect_language_per_element=detect_language_per_element,
         ),
     )
 
+    return elements
```

Comment:
A form like this might be worth considering. Same semantics, just maybe easier to follow in step-wise bite-sized pieces:

```python
    elements = _DocxPartitioner.iter_document_elements(
        filename,
       file,
        metadata_filename,
        include_page_breaks,
        metadata_last_modified,
    )
    elements = apply_lang_metadata(
        elements=elements,
        languages=languages,
        detect_language_per_element=detect_language_per_element,
    )
    return list(elements)
```

This also has the advantage that it's readily scalable, like if we add an additional step it can just get in line at the right place.

(same comment later on pptx partitioner)